modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/deep/pagination/configuration/DeepPaginationConfigurationWrapper.java

```diff
@@ -32,7 +32,28 @@
 )
 public class DeepPaginationConfigurationWrapper {
 
-	public boolean getEnableDeepPagination() {
+	public DeepPaginationConfiguration getDeepPaginationConfiguration(
+		long companyId) {
+
+		try {
+			if (companyId > CompanyConstants.SYSTEM) {  //this will not work, improve logic
+
+				return _configurationProvider.getCompanyConfiguration(
+					DeepPaginationConfiguration.class, companyId);
+			}
+
+			return _configurationProvider.getSystemConfiguration(
+				DeepPaginationConfiguration.class);
+		}
+		catch (ConfigurationException configurationException) {
+			return ReflectionUtil.throwException(configurationException);
+		}
+	}
+
+	public boolean getEnableDeepPagination(long companyId) {
```

Comment:
I would renamce to enableDeepPagination or maybe to more 'standard' isEnableDeepPagination. 'get' is usually not used to 'get' booleans.

Taking a look at https://github.com/liferay/liferay-portal/blob/99262f8fa6714886eca405ef5e0e5c407327e194/modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/configuration/ElasticsearchConfigurationWrapper.java#L61-L67 would speak for 'enableDeepPagination'